pkg/testutils/identity/allocator.go

```diff
@@ -74,8 +72,16 @@ func (f *MockIdentityAllocator) AllocateIdentity(_ context.Context, lbls labels.
 		return reservedIdentity, false, nil
 	}
 
+	requiresGlobal := identity.RequiresGlobalIdentity(lbls)
```

Comment:
The reason why I added this was to simplify unit testing in the following code block in `pkg/ipcache/metadata_test.go`:

```go
// Simulate kube-apiserver policy + CIDR policy on same prefix. Validate
// that removing the kube-apiserver policy will result in a new CIDR
// identity for the CIDR policy.
                                                                                
delete(identityMetadata, "1.1.1.1") // clean slate first
// Entry with only kube-apiserver labels means kube-apiserver is outside of
// the cluster, and thus will have a CIDR identity when InjectLabels() is
// called.
UpsertMetadata("1.1.1.1", labels.LabelKubeAPIServer)
assert.NoError(t, InjectLabels(source.Local, &mockUpdater{}, &mockTriggerer{}))
id := IdentityAllocator.LookupIdentityByID(
	context.TODO(),
	identity.LocalIdentityFlag, // we assume first local ID
)
assert.NotNil(t, id)
```

Essentially, it helps with validating that when we inject **only** the `kube-apiserver` label for a prefix into the IDMD, we get back a CIDR identity.  With this change, we allow the mock implementation to actually behave like the real allocator, without dependency on etcd, etc.

If this is problematic, then happy to discuss alternatives that allows us to validate label injection logic without having to fallback to an e2e test. As far as unit tests, this was the approach that seemed most reasonable to me.

Reply:
I guess in a way you could say that this unit test is reaching out into another package's purview (identity), so it's not exactly the most isolated, proper unit test.

Reply:
This unit test already has the IPIdentityCache, couldn't it look up the identity by IP in the ipcache and then check the reference counts on that? Then it doesn't matter what the identity itself is, the test only needs to verify the number of references held by the identity that has the IP->labels association...?

This case probably isn't a big deal, since there's a lot of assumptions in the codebase baked around this global/local identity split and I don't anticipate it changing any time soon. So I guess, if there's an easy simplification I'd be all for it. Otherwise this probably won't cause much problems in future (eg if refactoring the codebase).

Reply:
> This unit test already has the IPIdentityCache, couldn't it look up the identity by IP in the ipcache and then check the reference counts on that? Then it doesn't matter what the identity itself is, the test only needs to verify the number of references held by the identity that has the IP->labels association...?

Actually no it couldn't because the ipcache methods only act on `ipcache.Identity` rather than `identity.Identity`, which actually contains the reference count field, while the former does not.